utils/common/src/main/java/org/apache/brooklyn/util/time/DurationPredicates.java

```diff
@@ -0,0 +1,119 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.util.time;
+
+import com.google.common.base.Predicate;
+import com.google.common.base.Stopwatch;
+
+public class DurationPredicates {
+
+    /**
+     * @return A {@link Predicate} that checks if the {@link Duration} supplied
+     *         to {@link Predicate#apply(Object)} is positive.
+     */
+    public static Predicate<Duration> isPositive() {
+        return new Predicate<Duration>() {
+            @Override
+            public boolean apply(Duration duration) {
+                return duration != null && duration.isPositive();
+            }
+        };
+    }
+
+    /**
+     * @return A {@link Predicate} that checks if the {@link Duration} supplied
+     *         to {@link Predicate#apply(Object)} is negative.
+     */
+    public static Predicate<Duration> isNegative() {
+        return new Predicate<Duration>() {
+            @Override
+            public boolean apply(Duration duration) {
+                return duration != null && duration.isNegative();
+            }
+        };
+    }
+
+    /**
+     * @param duration
+     *            The {@link Duration} that will be the basis for comparison in
+     *            the returned {@link Predicate}.
+     * @return A {@link Predicate} that checks if the {@link Duration} that was
+     *         supplied to this method is longer than any {@link Duration}
+     *         supplied to to the returned {@link Predicate#apply(Object)}.
+     */
+    public static Predicate<Duration> isLongerThanDuration(final Duration duration) {
+        return new Predicate<Duration>() {
+            @Override
+            public boolean apply(Duration other) {
+                return duration != null && other != null && duration.isLongerThan(other);
+            }
+        };
+    }
+
+    /**
+     * @param duration
+     *            The {@link Duration} that will be the basis for comparison in
+     *            the returned {@link Predicate}.
+     * @return A {@link Predicate} that checks if the {@link Duration} that was
+     *         supplied to this method is longer than any {@link Stopwatch}
+     *         supplied to the returned {@link Predicate#apply(Object)}.
+     */
+    public static Predicate<Stopwatch> isLongerThanStopwatch(final Duration duration) {
+        return new Predicate<Stopwatch>() {
+            @Override
+            public boolean apply(Stopwatch other) {
+                return duration != null && other != null && duration.isLongerThan(other);
```

Comment:
This is confusing. A `Predicate<Stopwatch>` is making a boolean statement about `Stopwatch`-es, so in this case maybe reverse the check and the name?